pkg/tracegen/templated.go

```diff
@@ -82,6 +94,73 @@ type TraceTemplate struct {
 	Spans []SpanTemplate `js:"spans"`
 }
 
+type Link struct {
+	// LinkToPreviousSpanIndex true will set TraceID and SpanID the same as the previous span
+	LinkToPreviousSpanIndex bool `js:"linkToPreviousSpanIndex"`
+	// Attributes for this link
+	Attributes map[string]interface{} `js:"attributes"`
+	// Generate random attributes for this link
+	RandomAttributes *AttributeParams `js:"randomAttributes"`
+}
+
+type Event struct {
+	// Name of event
+	Name string `js:"name"`
+	// Attributes for this event
+	Attributes map[string]interface{} `js:"attributes"`
+	// Generate random attributes for this event
+	RandomAttributes *AttributeParams `js:"randomAttributes"`
+}
+
+type LinkDefaults struct {
+	// LinkToPreviousSpanIndex true will set TraceID and SpanID the same as the previous span
+	// Unless it is the first span then a random TraceID and a random SpanID will be used
+	LinkToPreviousSpanIndex bool `js:"linkToPreviousSpanIndex"`
```

Comment:
I'm wondering if we could make `true` the default for all links and remove this option. The reason is that this option is very specific and does not leave much room for future changes in the linking behavior without making this option obsolete. Maybe we want to link across traces in the future or link to random siblings in the same trace ...

Reply:
agreed. will just make it link to the previous span as the default until we can support linking to specific trace/span in the future